diff -urN octave-forge-2004.02.12.orig/octinst.sh.in octave-forge-2004.02.12/octinst.sh.in
--- octave-forge-2004.02.12.orig/octinst.sh.in	2004-03-25 21:09:35.301179072 +0100
+++ octave-forge-2004.02.12/octinst.sh.in	2004-03-25 21:09:23.130029368 +0100
@@ -18,8 +18,12 @@
 mpath=$1; shift
 opath=$1; shift
 xpath=$1; shift
-altmpath=$1; shift
-altopath=$1; shift
+if test $# -ge 1; then
+	altmpath=$1; shift
+	if test $# -ge 1; then
+		altopath=$1; shift
+	fi
+fi
 INSTALL="@INSTALL@"
 INSTALL_DATA="@INSTALL_DATA@"
 INSTALL_PROGRAM="@INSTALL_PROGRAM@"
