--- octave-forge-2004.02.12/main/geometry/configure.add.orig	2002-04-29 16:19:51.000000000 +0200
+++ octave-forge-2004.02.12/main/geometry/configure.add	2004-08-03 23:07:21.222218386 +0200
@@ -13,7 +13,6 @@
 cat > conftest.c <<EOF
 #include <stdio.h>
 #include <qhull/qhull.h>
-char qh_version[] = "version";
 int 
 main(argc, argv)
   int argc;
--- octave-forge-2004.02.12/main/geometry/convhulln.cc.orig	2003-12-14 17:53:48.000000000 +0100
+++ octave-forge-2004.02.12/main/geometry/convhulln.cc	2004-08-03 23:17:30.721723810 +0200
@@ -29,7 +29,6 @@
 #endif
 #include <octave/oct.h>
 
-char qh_version[] = "convhulln.oct 2003-12-14";
 char flags[250];
 const char *options;
 
--- octave-forge-2004.02.12/main/geometry/__voronoi__.cc.orig	2003-12-14 17:53:48.000000000 +0100
+++ octave-forge-2004.02.12/main/geometry/__voronoi__.cc	2004-08-03 23:18:01.095418937 +0200
@@ -35,7 +35,6 @@
 #endif
 #include <octave/oct.h>
 
-char qh_version[] = "__voronoi__.oct 2003-12-14";
 FILE *outfile = stdout;
 FILE *errfile = stderr;
 char flags[250];
--- octave-forge-2004.02.12/main/geometry/delaunayn.cc.orig	2004-01-24 18:42:18.000000000 +0100
+++ octave-forge-2004.02.12/main/geometry/delaunayn.cc	2004-08-03 23:18:38.666383170 +0200
@@ -39,7 +39,6 @@
 #endif
 #include <octave/oct.h>
 
-char qh_version[] = "delaunayn.oct 2003-12-14";
 FILE *outfile = stdout;
 FILE *errfile = stderr;
 char flags[250];
