--- octave-forge-2004.02.12/extra/Makefile.orig	2003-08-25 15:11:38.000000000 +0000
+++ octave-forge-2004.02.12/extra/Makefile	2004-08-02 20:53:55.000000000 +0000
@@ -35,7 +35,7 @@
 $(SUBMAKEDIRS):
 	@if test -z "$(MAKECMDGOALS)" ; then \
 	    cd $@ && $(MAKE) ; \
-	elif grep "^$(MAKECMDGOALS) *[:]" $@Makefile >/dev/null; then \
+	elif grep "^$(MAKECMDGOALS) *[:]" $@/Makefile >/dev/null; then \
 	    cd $@ && $(MAKE) $(MAKECMDGOALS) ; \
 	fi
 
--- octave-forge-2004.02.12/main/Makefile.orig	2002-12-31 18:43:04.000000000 +0000
+++ octave-forge-2004.02.12/main/Makefile	2004-08-02 20:56:28.000000000 +0000
@@ -35,7 +35,7 @@
 $(SUBMAKEDIRS):
 	@if test -z "$(MAKECMDGOALS)" ; then \
 	    cd $@ && $(MAKE) ; \
-	elif grep "^$(MAKECMDGOALS) *[:]" $@Makefile >/dev/null ; then \
+	elif grep "^$(MAKECMDGOALS) *[:]" $@/Makefile >/dev/null; then \
 	    cd $@ && $(MAKE) $(MAKECMDGOALS) ; \
 	fi
 
